.github/workflows/main.yml

```diff
@@ -335,7 +907,9 @@ jobs:
       - name: Test PHP Doctrine2
         run: ant $OPTS -f php/php.doctrine2 test
 
+      # todo: longest step -> needs 2 tiers, with tier 2 conditional
       - name: Test PHP Editor
+        if: contains(github.event.pull_request.labels.*.name, 'PHP') || github.event_name != 'pull_request'
         run: ant $OPTS -f php/php.editor test
```

Comment:
@neilcsmith-net this seems to work. The step runs only if the PR has the PHP label on it. If this isn't a PR, it will run too (e.g merge to master). This alone would cut down 2x40mins per sync and the fast steps would still run. Ideally someone would split php.editor into two tiers some time in future.

Requires to label the PRs correctly. Other labels could follow if more categories with long running jobs are found.

This is somewhat similar to how the OpenJDK project works, its just more sophisticated there (a bot manages everything and sets things).

We could also add a full-test label, so that if someone isn't sure or changes something in the platform or API, it would just turn everything on.

Reply:
just checked: 
labels are only picked up once when the workflow starts. Restarting jobs won't update them. A sync would trigger a new workflow with updated labels though. Correct labeling would be required *before* creating a PR to conserve resources.

I added a ci:no-build label for PRs which only update readmes etc.

Reply:
What are the pros / cons of running by label vs running by path?

Reply:
path triggers are per-workflow, not per-job. The pipeline does only work within a workflow where artifacts can be shared. So we would have to implement our own condition script which checks paths somehow to conditionally run jobs.

but the main difference: devs set labels. If a reviewer wants all tests to run it could be easily set. Or if tier1 is fine only fast tests are run.

Automating this does rarely work. Calculating the paths would basically trigger everything all the time, see https://github.com/apache/netbeans/pull/3727#issuecomment-1061752817
And what it it doesn't trigger but you want it to trigger? You still need some fallback.

Reply:
btw I don't think we should label trigger everything, only jobs which are long enough. The profiler job runs in <2 mins, there is not much to be gained to conditionally run this.

Reply:
OK, makes sense.  Slight concern at our proliferation of labels and making sure it's well documented so there aren't too many surprises.  Agree automating has issues, although if we want it later we might always automate adding labels! :smile:

Reply:
@neilcsmith-net 197e209, 4 category labels so far, for `Java`, `JavaScript`, `Platform` and `PHP`. The label `ci:all-tests` will cause everything to run, `ci:no-build` cancels everything, e.g thats for fixing typos in readmes etc.